src/models/member.ts

```diff
@@ -25,5 +27,9 @@ export class Member {
     if ('is_org_admin' in data) {
       this.isOrgAdmin = data['is_org_admin'];
     }
+
+    if ('user_details' in data) {
+      this.userDetails = new User(data['user_details']);
```

Comment:
You don't need both a user and a userDetail. Change the this.user field to be a User() model, and use data.user and data.user_details to populate it.